src/algebra/group_with_zero.lean

```diff
@@ -350,6 +350,12 @@ lemma mul_left_inj' (hc : c ≠ 0) : a * c = b * c ↔ a = b := ⟨mul_right_can
 
 lemma mul_right_inj' (ha : a ≠ 0) : a * b = a * c ↔ b = c := ⟨mul_left_cancel' ha, λ h, h ▸ rfl⟩
 
+@[simp] lemma mul_eq_mul_right_iff : a * c = b * c ↔ a = b ∨ c = 0 :=
+by by_cases hc : c = 0; [simp [hc], simp [mul_left_inj', hc]]
```

Comment:
What would be the effect of just marking `mul_right_inj'` `@[simp]` instead? Would `simp` introduce separate goals `a ≠ 0` and `b = c`? (arguably easier to work with).

Reply:
`mul_left_inj'` is not an `iff`.

Reply:
E.g., `mul_left_inj'` doesn't help me solve `2 * m = m * m` without a manual intervention.